### DISCUSSION TITLE
Why findFirst and findUnique types are not `T | null`

### DISCUSSION THREAD

**Discussion by incompletude** [prisma/prisma#10075] [Q&A]
In the example bellow, the `user` type is `User`. Why isn't  it `User | null`?

```
    const user = await this.prismaService.user.findFirst({
      where: {
        field: 1,
      },
    })
```

For a type safe ORM this seems very weird.

**Comment by aqrln** [comment]
Hi @incompletude! It is, in fact, `User | null`. The problem is most likely in your configuration: unless you have `"strict": true` or at least `"strictNullChecks": true` in `"compilerOptions"` your `tsconfig.json`, TypeScript will "simplify" `User | null` to just `User`.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
